src/python/pants/backend/project_info/list_roots.py

```diff
@@ -52,9 +58,9 @@ async def all_roots(source_root_config: SourceRootConfig) -> AllSourceRoots:
 
 
 @goal_rule
-async def list_roots(console: Console, options: RootsOptions, all_roots: AllSourceRoots) -> Roots:
+async def list_roots(console: Console, options: RootsOptions, asr: AllSourceRoots) -> Roots:
```

Comment:
This change seems worse. `all_roots` was good to begin with.

Reply:
It shaded the name of the `all_roots` rule, so IDE checkers warn about it.